.maintain/add-prepush-hook.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+# write a whole script to pre-push hook
+# NOTE: it will overwrite pre-push file!
+cat > .git/hooks/pre-push <<'EOF'
+#!/bin/bash
+
+echo "Running clippy checks"
+
+set +e
+
+cargo clippy --all-features -- -W clippy::all -A clippy::style -A clippy::forget_copy -A clippy::forget_ref
```

Comment:
Let's add both checks we have in the ci workflow, ie. [this](https://github.com/pendulum-chain/spacewalk/blob/bda384bcc6db3da88198c714851a261dcdea1063/.github/workflows/ci-main.yml#L84) and [this](https://github.com/pendulum-chain/spacewalk/blob/bda384bcc6db3da88198c714851a261dcdea1063/.github/workflows/ci-main.yml#L91) one.